libunwind/src/CompartmentInfo.hpp

```diff
@@ -13,21 +13,68 @@
 #ifndef __COMPARTMENT_INFO_HPP__
 #define __COMPARTMENT_INFO_HPP__
 
+extern "C" {
+
+#include <link.h>
```

Comment:
Since the declarations for `dl_c18n_is_tramp` and `dl_c18n_pop_trusted_stk` are now in `link_elf.h`, I include `link.h` to avoid having to re-declare both functions.

Reply:
Ok, but then you need to detect them in CMake, because otherwise you'll break the build on Linux or current/old versions of CheriBSD.

Reply:
Also, why isn't the struct declared by system headers then?..

Reply:
Hmmm maybe it's easier to just hard-code everything here in libunwind then.

Reply:
I added some detection logic in CMake and a `#define` in CheriBSD's config. Not sure if this matches what you have in mind.

Reply:
Well, test it? See if you can build against old and new CheriBSD.

Reply:
And make sure to test against CHERI-RISC-V too.

Reply:
OK. I've successfully built the latest commit with CTSRD-CHERI/cheribsd#2123 for both Morello and CHERI-RISC-V, and it also builds on an older CheriBSD installation through CMake.

Reply:
And when built for Morello with 2123 it detected c18n support?